static/app/components/dataExport.tsx

```diff
@@ -70,11 +71,14 @@ export function useDataExport({
         if (unmountedRef?.current) {
           return;
         }
+        const isSelfHosted = ConfigStore.get('isSelfHosted');
         const message =
-          err?.responseJSON?.detail ??
-          t(
-            "We tried our hardest, but we couldn't export your data. Give it another go."
-          );
+          err?.status === 404 && !isSelfHosted
+            ? t('Data export requires Business Plan')
```

Comment:
shouldn't the button be disabled or removed if they can't export on their plan

Reply:
yeah that is probably the better fix

Reply:
i just updated the branch. let me know if that makes sense.